### DISCUSSION TITLE
XML-Compare

### DISCUSSION THREAD

**Discussion by LeoKurz68** [WinMerge/winmerge#2417] [Ideas]
Hello, 
I often have to compare XML-files with almost same content but every time differently ordered in every layer which makes comparison merley impossible. It would be extremely helpfull, if winmerge could sort/compare by content, not by the order in the file. While searching the internet, I have found a lot of people looking for the same functionality but nnothing that really does the trick. You can find quite a lot of code  for programming etc. but not an actual application. As I love and use winmerge a lot, the intgegartion in this app would be phenomenal  :-)
__Leo

**Comment by sdottaka** [comment]
Unfortunately, WinMerge does not provide a function to sort XML files.

However, you can create a plugin to sort XML files.

If you have python installed, you can create a plugin as follows.

1. Install lxml as follows:
~~~
pip install lxml
~~~
2. Place sortxmlpy.py included in the attached file in c:\tmp\.
(This sortxmlpy.py sorts XML files by element name)

3. Define the plugin in the Plugin window that appears when you select the Plugin → Plugin Settings menu item in WinMerge as follows.

![image](https://github.com/user-attachments/assets/f3cef7f2-6b3f-41ac-9937-df347867d762)

The plugin is applied as follows:

![image](https://github.com/user-attachments/assets/86330b7a-7dd4-472d-9fe3-583dd5cb3d27)
![image](https://github.com/user-attachments/assets/31151167-ed8a-40e8-9db6-a74fabe2b629)

[sortxmlpy.zip](https://github.com/user-attachments/files/16808228/sortxmlpy.zip)
